airbytehq/airbyte
# feat(airbyte-ci): Enable manifest-only connector unit tests in CI

## What

Enables unit testing capabilities for manifest-only connectors and validates the approach using `source-the-guardian-api` as a proof of concept. This change allows manifest-only connectors to maintain test coverage of custom components while benefiting from the simplified manifest-based architecture.

Demo: https://www.loom.com/share/80b3ae7282e94dd0acfa34854b877462?sid=18b54fd6-79e0-4154-83ba-462d4345ddcb

### Note

The changes in this PR are reliant on the availability of the test fixtures in the CDK. The PR to include those fixtures can be found here: https://github.com/airbytehq/airbyte-python-cdk/pull/121

## How
 
### Guardian API PoC:
- Implemented unit tests for the source's CustomPageIncrement component
- Set the base_image version to a release candidate of the CDK containing the fixtures needed to dynamically load `components.py`

### airbyte-ci
- Modified `airbyte-ci` to support running unit tests for manifest-only connectors:
- Added `ManifestConnectorUnitTests` class inheriting from `PytestStep`
- Enabled unit test execution during CI

This approach provides a bridge for maintaining test coverage as we transition connectors to the manifest-only format, ensuring we can continue to maintain existing unit tests for custom Python components.

## How to Test

1. Pull this branch locally and navigate to the `airbyte-integrations/connectors/source-the-guardian-api/unit_tests` folder
2. Run `poetry install` and `poetry run pytest` to verify the tests run locally.
3. Navigate to `airbyte-ci/connectors/pipelines` and run `poetry install` and `poetry shell`
4. Run `airbyte-ci connectors --name=source-the-guardian-api test --only-step=unit` to run just the unit test suite against a built connector image.
5. The results of the full test run can also be verified in the CI checks for this PR.


## Can this PR be safely reverted and rolled back?
